course3/UniquePtr/my_unique_ptr.h

```diff
@@ -0,0 +1,221 @@
+#include <iostream>
+/*
+re-implemention of http://en.cppreference.com/w/cpp/memory/unique_ptr
+*/
+
+template<class T>
+class MyUniquePtr
+{
+public:
+    MyUniquePtr() = default;
+    
+    ~MyUniquePtr(){}
```

Comment:
The sole goes of unique_ptr is to destroy the pointed-to object at the end of it's lifetime.